processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -102,6 +110,20 @@ public TypeFactory(Elements elementUtils, Types typeUtils) {
         implementationTypes.put( ConcurrentNavigableMap.class.getName(), getType( ConcurrentSkipListMap.class ) );
     }
 
+    private static List<AstModifyingAnnotationProcessor> getAstModifyingAnnotationProcessors() {
+        List<AstModifyingAnnotationProcessor> processors = new ArrayList<AstModifyingAnnotationProcessor>();
+
+        ServiceLoader<AstModifyingAnnotationProcessor> loader = ServiceLoader.load(
+                AstModifyingAnnotationProcessor.class
```

Comment:
Make this:

    ServiceLoader.load(AstModifyingAnnotationProcessor.class, AstModifyingAnnotationProcessor.class.getClassLoader());

Without the second param, you get the thread context loader and that's not the right one (all the APs share a special loader and we need that one).